CatsAndDogs/ContentLoader.swift

```diff
@@ -16,11 +16,18 @@ private extension Data {
 }
 
 class OnlineContentLoader {
-    let url = URL(string: "https://catsanddogs-kotlin-bff.herokuapp.com/schedule.json?from=2017-05-08T12:00:00-02:00")
+    let urlString = "https://catsanddogs-kotlin-bff.herokuapp.com/kotlinconf/schedule.json"
     let session = URLSession(configuration: .default)
 
     private func downloadSchedule(completion: @escaping (Data?)->()) {
-        session.dataTask(with: url!) { (data, _, _) in completion(data) }.resume()
+        if let url = URL(string: urlString) {
+            var request = URLRequest(url: url)
```

Comment:
I find it strange that it must be a var, the reference itself is not mutable. In kotlin it could be a `val`, even if you can add headers

Reply:
Oh that's a `mutating` function, interesting! We don't have that concept in kotlin. Nice read around that: https://www.natashatherobot.com/mutating-functions-swift-structs/

Reply:
Yeah, structs are immutable by default so this opts you out of that. Since they are value types you don't want to encourage mutation as you don't get the expected behaviour that some people might think (change one place, it doesn't change in another) but it makes sense in some cases, like this.